graph/api/src/main/java/org/jboss/windup/graph/model/ProjectModelTraversal.java

```diff
@@ -0,0 +1,107 @@
+package org.jboss.windup.graph.model;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.windup.graph.model.resource.FileModel;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+
+/**
+ * This allows a {@link ProjectModel} to be traversed in a way that is aware of {@link DuplicateProjectModel}s.
+ *
+ * This should make it easier to calculate the actual path within the application without regard to where the original
+ * project was actually stored.
+ *
```

Comment:
Could you please add something along the lines we were talking about this - i.e. "this is for cases like Java application details, when you need to traverse through the actual files of it, rather than the canonical archives/projects" ? Thx